### ISSUE TITLE
Accept host id on create template

### ISSUE THREAD

**Issue by jdotcms** [dotCMS/core#21564]
Currently the template is stored on the current host, however the rest client should has the ability to send the host id in order to store the template on a particular host (optional value)

**Comment by bryanboza** [comment]
Fixed, tested on release-22.03 // Docker // FF

**Comment by wezell** [comment]
Example of how you post an advanced template

```
curl --location -g --request POST '{{serverURL}}/api/v1/templates' \
--header 'Authorization: Basic YWRtaW5AZG90Y21zLmNvbTphZG1pbg==' \
--header 'Content-Type: application/json' \
--data-raw '{
    "siteId":"{{siteId}}",
    "title":"PostMan Test",
    "friendlyName":"This is a postman test description",
    "body":"This is the body"
}'

```
